vendor/github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/framework/machines.go

```diff
@@ -63,22 +63,28 @@ func (f *Framework) DeleteMachineAndWait(machine *machinev1beta1.Machine, client
 	}
 }
 
-func (f *Framework) waitForMachineToRun(machine *machinev1beta1.Machine, client types.CloudProviderClient) {
+func (f *Framework) waitForMachineToRun(machine *machinev1beta1.Machine, client types.CloudProviderClient) *machinev1beta1.Machine {
 	f.By(fmt.Sprintf("Waiting for %q machine", machine.Name))
+	var updatedMachine *machinev1beta1.Machine
 	// Verify machine has been deployed
 	err := wait.Poll(PollInterval, TimeoutPoolMachineRunningInterval, func() (bool, error) {
-		if _, err := f.CAPIClient.MachineV1beta1().Machines(machine.Namespace).Get(machine.Name, metav1.GetOptions{}); err != nil {
+		var err error
+		if updatedMachine, err = f.CAPIClient.MachineV1beta1().Machines(machine.Namespace).Get(machine.Name, metav1.GetOptions{}); err != nil {
 			glog.V(2).Infof("Waiting for '%v/%v' machine to be created", machine.Namespace, machine.Name)
 			return false, nil
 		}
+		if updatedMachine.Spec.ProviderID == nil {
```

Comment:
There is no guarantee `ProviderID` field will be set. E.g. libvirt still does not set the field. This will break actuators that do not set the field.

Plus the fact a machine (its underlying instance) is running has nothing to do with `ProviderID` field being set.

Reply:
handled!